runtime/jcl/common/reflecthelp.c

```diff
@@ -1720,3 +1720,202 @@ getFieldsHelper(JNIEnv *env, jobject cls)
 	vmFuncs->internalExitVMToJNI(vmThread);
 	return result;
 }
+
+static j9object_t
```

Comment:
Can you doc this function as well?

Is there a relationship between the accessors and the order they get generated in?  Can we probe the `i`th method as a first check when looking at the `i`th component?  Or does javac not generate them in a particular order?

Reply:
yes this can be optimized.